### PR TITLE
macOS Xcode Project Updates to XCFrameworks C++17 - Bleeding Edge

### PR DESCRIPTION
**Pull request by danoli3**
macOS / OSX Bleeding Edge Update

Built with latest Binaries 
Screenshot 2024-02-08 at 12 22 14 am

Presenting Apple XCFrameworks 
- macOS initial project changes for latest bleeding edge libraries and formatted in xcframeworks format.

XCFrameworks provide libraries in structured formats that allows for multiple architectures, targets all within the same project without causing linking errors and does provide target specific headers

Extra information to be packaged with each binary is a unique hash, framework version number, build date, checksums of binaries. With the latest changes comes sweeping security and vulnerability patches beyond count. 

iOS / tvOS / MacCatalyst to follow soon

### CODE REVIEW

**Comment by ofTheo** [comment]
Sweet! 

Looks like ( from the checks ) it is missing the libs though. 
Would this need to pull from bleeding apothecary ?

**Comment by danoli3** [comment]
Just some potions to refine first and yeah i'll update github actions update and pull new download script

**Comment by danoli3** [comment]
Updated with linking to latest libraries / scripts now.

Validated Project Generator working
Validated Project Generator working for addons with xcframework (opencv)

**Comment by danoli3** [comment]
Okay fixed include copy using standard paths was not copying correctly from CI! 

Should be ready for test @ofTheo

**Comment by danoli3** [comment]
Xcode all working disregard make file stuff for now

**Comment by danoli3** [comment]
Awesome fixed the make files, they were actually more sensitive to real problems so patched libraries correctly now! 
Final potion fixes: 
- OpenCV no Nvidia carotene on arm64 macOS 
- Fixed FreeImage sub dependency LibPNG - now links to built lib rather than outdated package - solves double symbol and security issues

**Comment by dimitre** [comment]
I think there is a typo here : 
```HEADER_FMT = "$(OF_PATH)/libs/glm/include"```
glm -> fmt

**Comment by danoli3** [comment]
Thanks! Yeah fixed that although I think need to test if headers are now automatically included from xcframework, I think maybe 

Merged in https://github.com/openframeworks/openFrameworks/pull/7899 into this PR now. Makes sense to. Closing that one. 

tvOS / iOS Projects / now linking against super mega xcframework and compiling successfully / running

**Comment by ofTheo** [comment]
Awesome!

If all the checks are passing and not due to caching or checks being skipped I am fine to merge this. 
Be good to get a look at from @dimitre too.

**Comment by danoli3** [comment]
Yeah this is looking good to go! Yeah tests are all working and building locally too. 

I do have further Mega Project stuff to Merge after this PR, however that one includes a lot of Defines as the whole project context is available for all platforms, in a workplace style within the macos folder, so best if separated after this PR

**Comment by ofTheo** [comment]
Sorry for the delay getting back to this @danoli3 
If you want to resolve the conflicts - we can merge :)

**Comment by danoli3** [comment]
Rebased!

**Comment by danoli3** [comment]
Let me fix, I think one of the rebases I took wrong head

File: libs/openFrameworks/utils/ofURLFileLoader.cpp [hunk 1]
```diff
@@ -275,6 +275,12 @@ ofURLFileLoader::ofURLFileLoader()
 	: impl(new ofxEmscriptenURLFileLoader) { }
 #endif
 
+#ifdef TARGET_OF_TVOS
+#include "ofxtvOSURLFileLoader.h"
+ofURLFileLoader::ofURLFileLoader()
+: impl(new ofxtvOSURLFileLoader) { }
```
Comment (danoli3) [inline libs/openFrameworks/utils/ofURLFileLoader.cpp:281]: need to fix this later post curl fixes

File: libs/openFrameworksCompiled/project/ios/iOS+OFLib.xcodeproj/project.pbxproj
Comment (danoli3) [inline libs/openFrameworksCompiled/project/ios/iOS+OFLib.xcodeproj/project.pbxproj:1]: iOS currently targets macOS mega framework

File: scripts/dev/xcode_projects_to_xml.sh
Comment (danoli3) [inline scripts/dev/xcode_projects_to_xml.sh:1]: we using json now actually might delete this

**Review by danoli3 — COMMENTED** [review]
This is looking pretty ready

**Comment by danoli3** [comment]
Leeroy

**Pull request merged**